src/utils/DateUtils.ts

```diff
@@ -29,138 +157,230 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
+    DateUtils.setLocale();
+
+    if (Utils.isNullOrUndefined(date)) {
       return '';
     }
 
     options = new DefaultDateToStringOptions().merge(options);
 
-    var dateOnly = DateUtils.keepOnlyDatePart(d);
+    const dateOnly = moment(DateUtils.keepOnlyDatePart(date));
+    const today = moment(DateUtils.keepOnlyDatePart(options.now));
 
     if (options.predefinedFormat) {
-      return Globalize.format(dateOnly, options.predefinedFormat);
+      // yyyy was used to format dates before implementing moment.js, which only recognizes YYYY.
+      return dateOnly.format(options.predefinedFormat.replace(/yyyy/g, 'YYYY'));
     }
 
-    var today = DateUtils.keepOnlyDatePart(options.now);
+
     if (options.useTodayYesterdayAndTomorrow) {
-      if (dateOnly.valueOf() == today.valueOf()) {
-        return l('Today');
-      }
-      var tomorrow = DateUtils.offsetDateByDays(today, 1);
-      if (dateOnly.valueOf() == tomorrow.valueOf()) {
-        return l('Tomorrow');
-      }
-      var yesterday = DateUtils.offsetDateByDays(today, -1);
-      if (dateOnly.valueOf() == yesterday.valueOf()) {
-        return l('Yesterday');
+      if (DateUtils.isTodayYesterdayOrTomorrow(date, options)) {
+        return moment(dateOnly).calendar(moment(today));
       }
     }
 
-    var isThisWeek = Math.abs(TimeSpan.fromDates(dateOnly, today).getDays()) < 7;
+    const isThisWeek = dateOnly.diff(moment(today), 'weeks') == 0;
+
     if (options.useWeekdayIfThisWeek && isThisWeek) {
       if (dateOnly.valueOf() > today.valueOf()) {
-        return l('NextDay', Globalize.format(dateOnly, 'dddd'));
+        return l('NextDay', dateOnly.format('dddd'));
+      } else if (dateOnly.valueOf() < today.valueOf()) {
+        return l('LastDay', dateOnly.format('dddd'));
       } else {
-        return l('LastDay', Globalize.format(dateOnly, 'dddd'));
+        return dateOnly.format('dddd');
       }
     }
 
-    if (options.omitYearIfCurrentOne && dateOnly.getFullYear() === today.getFullYear()) {
-      return Globalize.format(dateOnly, 'M');
+    if (options.omitYearIfCurrentOne && dateOnly.year() === today.year()) {
+      return dateOnly.format('MMMM DD');
     }
 
     if (options.useLongDateFormat) {
-      return Globalize.format(dateOnly, 'D');
+      return dateOnly.format('dddd, MMMM DD, YYYY');
     }
 
-    return Globalize.format(dateOnly, 'd');
+    return dateOnly.format('M/D/YYYY');
   }
 
+  /**
+   * Converts the time information from a date into a string.
+   * @param date The date containing the time information to format.
+   * @param options Specifies the enabled options on the date.
+   * @returns {any} Will either be an empty string or a string containing the time.
+   */
   static timeToString(date: Date, options?: IDateToStringOptions): string {
     if (Utils.isNullOrUndefined(date)) {
       return '';
     }
-
-    return Globalize.format(date, 't');
+    return moment(date).format('h:mm A');
   }
 
```

Comment:
> Creates a string from a Date object. The resulting string is formatted according to a set of options. This method calls [`timeToString`]{@link DateUtils.timeToString} to add time information to the date. If you need to create a date string without a timestamp, use the [`dateToString`]{@link DateUtils.dateToString} method instead.
> @param date The Date object to create a string from.
> @param options The set of options to apply when formatting the resulting string. If you do not specify a value for this parameter, the method uses a default set of options.
> @returns {string} A date string corresponding to the `date` argument value, formatted according to the specified `options`. 

Note: I suggest changing the return type from "any" to "string" in the documentation. Please make sure this is correct.